woocommerce/sensei-certificates
# Update tFPDF version

Fixes #117 

This PR updates the version of the [tFPDF library](https://github.com/DocnetUK/tfpdf) from 1.24 to. 2.1.0 (the latest version).

Unfortunately, we have two methods in Sensei Certificates that use the tFPDF library with very similar code. Ideally, the common functionality should be extracted, but maybe in a different PR.

However, I did extract some functionality that was changed in the new version, to prevent further code duplication.

The meaningful changes are in `classes/*`. All of the changes in `lib/*` are from upgrading the tFPDF code.

## Concerns

The biggest concern is that the new version of tFPDF does not support PHP 5.3 or lower. If we want to maintain support in this extension for earlier versions of PHP, we may have to consider modifying the library to address the compatibility issues.

Also, Sensei Certificates appears to have had the ability to save a PDF to a file. When I tried to test this, it threw a Fatal and didn't work:

```
PHP Fatal error:  Uncaught Error: Call to undefined method WooThemes_Sensei_PDF_Certificate::get_certificate_template_path() in /var/www/html/wp-content/plugins/sensei-certificates/classes/class-woothemes-sensei-pdf-certificate.php:149
```

I removed this functionality, because we are not using it in our own extension, and it seems that no one else is using it because it is broken. That said, I wonder if we should be more cautious about removing it?

(Also note the `$path` parameter that I've left in the `generate_pdf` functions for potential backward compatibility.)

## Testing

- Create a certificate template and add it to a course.

- When viewing the template in wp-admin, click on the Preview button. You should see the PDF preview for the certificate template.

- Complete the course.

- View the certificates in wp-admin. You should see the certificate that you earned.

- Click on View Certificate. You should see the PDF of the certificate.
